indra/newview/llstartup.cpp

```diff
@@ -206,11 +215,704 @@
+                              << " current_version=" << current_child_version
+                              << LL_ENDL;
+            mFetchedCategories.insert(child_id);
+        }
+    }
+}
+
+void LLAsyncInventorySkeletonLoader::discoverEssentialFolders()
+{
+    static const LLFolderType::EType essential_types[] = {
+        LLFolderType::FT_CURRENT_OUTFIT,
+        LLFolderType::FT_MY_OUTFITS,
+        LLFolderType::FT_LOST_AND_FOUND,
+        LLFolderType::FT_TRASH,
+        LLFolderType::FT_INBOX,
+        LLFolderType::FT_OUTBOX
+    };
+
+    for (LLFolderType::EType type : essential_types)
+    {
+        LLUUID cat_id = gInventory.findCategoryUUIDForType(type);
+        if (cat_id.isNull())
+        {
+            continue;
+        }
+
+        if (type == LLFolderType::FT_CURRENT_OUTFIT)
+        {
+            mSawCurrentOutfitFolder = true;
+        }
+
+        LLViewerInventoryCategory* cat = gInventory.getCategory(cat_id);
+        bool is_library = false;
+        if (cat)
+        {
+            is_library = (cat->getOwnerID() == gInventory.getLibraryOwnerID());
+        }
+
+        const S32 cached_version = gInventory.getCachedCategoryVersion(cat_id);
+        if (cat && isCategoryUpToDate(cat, cached_version))
+        {
+            mFetchedCategories.insert(cat_id);
+            LL_INFOS("AsyncInventory") << "Essential folder up to date from cache"
+                              << " cat_id=" << cat_id
+                              << " name=\"" << cat->getName() << "\""
+                              << " cached_version=" << cached_version
+                              << " current_version=" << cat->getVersion()
+                              << " descendents=" << cat->getDescendentCount()
+                              << LL_ENDL;
+            continue;
+        }
+
+        if (mFetchedCategories.count(cat_id) == 0 && mQueuedCategories.count(cat_id) == 0 && mActiveFetches.count(cat_id) == 0)
+        {
+            enqueueFetch(cat_id, is_library, true, cached_version);
+            mEssentialPending.insert(cat_id);
+            LL_INFOS("AsyncInventory") << "Essential folder queued for fetch"
+                              << " cat_id=" << cat_id
+                              << " cached_version=" << cached_version
+                              << " current_version=" << (cat ? cat->getVersion() : LLViewerInventoryCategory::VERSION_UNKNOWN)
+                              << LL_ENDL;
+        }
+    }
+
+    LLUUID cof_id = LLAppearanceMgr::instance().getCOF();
+    if (cof_id.notNull()
+        && mFetchedCategories.count(cof_id) == 0
+        && mQueuedCategories.count(cof_id) == 0
+        && mActiveFetches.count(cof_id) == 0)
+    {
+        mSawCurrentOutfitFolder = true;
+        LLViewerInventoryCategory* cof = gInventory.getCategory(cof_id);
+        const S32 cached_version = gInventory.getCachedCategoryVersion(cof_id);
+        if (isCategoryUpToDate(cof, cached_version))
```

Comment:
Please add to test plan:
- Log in on machine A. Log out
- Wear an outfit with multiple items of the same type on machine B. Log out.
- Log in on machine A. Change outfit order. Log out.
- Log in on machine B. Confirm order of items.